FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationValidators/ConfigurationValidatorV1.cs

```diff
@@ -21,28 +21,53 @@ public static class ConfigurationValidatorV1
         /// <exception cref="ConfigurationErrorException">Throw ConfigurationErrorException if configuration is invalid.</exception>
         public static void Validate(IServiceCollection services)
         {
-            FhirServerConfiguration fhirServerConfiguration;
+            DataSourceConfiguration dataSourceConfiguration;
             try
             {
                 // include enum field, an exception will be thrown when parse invalid enum string
-                fhirServerConfiguration = services
+                // enum field parsing is case insensitive
+                dataSourceConfiguration = services
                     .BuildServiceProvider()
-                    .GetRequiredService<IOptions<FhirServerConfiguration>>()
+                    .GetRequiredService<IOptions<DataSourceConfiguration>>()
                     .Value;
             }
             catch (Exception ex)
             {
-                throw new ConfigurationErrorException("Failed to parse fhir server configuration", ex);
+                throw new ConfigurationErrorException("Failed to parse data source configuration", ex);
             }
 
-            if (string.IsNullOrWhiteSpace(fhirServerConfiguration.ServerUrl))
+            switch (dataSourceConfiguration.Type)
             {
-                throw new ConfigurationErrorException("Fhir server url can not be empty.");
-            }
+                case DataSourceType.FHIR:
```

Comment:
Maybe add the check, if fhir server configuration come from root **fhirServer** rather than **dataSource**, the **DataSourceType** must be FHIR?

Reply:
If customers specify `dataSource`, the root `fhirServer` section will ignored. In current design, we only look into `fhirServer` if `dataSource` is not found.